### DISCUSSION TITLE
How do I remap existing keymaps?

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#2256] [General]
E.g. I would like to remap <leader>fR to <leader>fr

Do I break anything if I just do it inside keymaps.lua?

**Comment by dpetka2001** [comment]
You probably mean `<leader>fR` if I'm not mistaken? The proper way to do what you want is to search the repo for the keymap you want and change it accordingly in your personal configuration. 

For example, `<leader>fR` is configured [here](https://github.com/LazyVim/LazyVim/blob/879e29504d43e9f178d967ecc34d482f902e5a91/lua/lazyvim/plugins/editor.lua#L144) in the `keys` section of the Telescope spec. The reason you should do it in the `keys` section of the Telescope spec, is because those keys are also used for lazy-loading Telescope. If you define the keymap you want in `keymaps.lua` you lose the ability of lazy-loading the plugin.

**Reply by idelice** [reply]
Ah ok. I added and modifed those two lines:

```
      { "<leader>fR", "<cmd>Telescope oldfiles<cr>", desc = "Recent" },
      { "<leader>fr", Util.telescope("oldfiles", { cwd = vim.loop.cwd() }), desc = "Recent (cwd)" },
```

and imported the util and it worked perfectly fine. Thank you

**Reply by dpetka2001** [reply]
Just to clarify you did that in your personal configuration right? You didn't change it in `~/.local/share/nvim/lazy/` directory, correct?

**Reply by idelice** [reply]
Correct

**Comment by rubiin** [comment]
I wanted to remap the code actions keybindings . "<leader>ca" and "<leader>cA" to my own implementation but the remaps only shows on dashboard but rolls back to lazyvims setting @dpetka2001

**Reply by dpetka2001** [reply]
Those are LSP keymaps and have a slightly different way of [customizing](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) them.

**Reply by rubiin** [reply]
also I tried disabling all keymaps but they are not working .

```lua
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    {
      "LazyVim/LazyVim",
      import = "lazyvim.plugins",
    },
    {
      "LazyVim/LazyVim",
    import = "lazyvim.plugins.core",
      opts = {
        defaults = {
          autocmds = false,
          keymaps = false,
        },
      },
  },
    -- import any extras modules here
    { import = "lazyvim.plugins.extras.lsp.none-ls" },
    -- languages
    { import = "lazyvim.plugins.extras.lang.docker" },
    { import = "lazyvim.plugins.extras.lang.go" },
    { import = "lazyvim.plugins.extras.lang.json" },
    { import = "lazyvim.plugins.extras.lang.markdown" },
    { import = "lazyvim.plugins.extras.lang.yaml" },
    { import = "lazyvim.plugins.extras.lang.tailwind" },
    { import = "lazyvim.plugins.extras.lang.typescript" },
    { import = "lazyvim.plugins.extras.lang.yaml" },
    -- testing and debugging
    { import = "lazyvim.plugins.extras.dap.core" },
    { import = "lazyvim.plugins.extras.test.core" },
    -- misc
    -- { import = "lazyvim.plugins.extras.formatting.prettier" }, --replace with biome
    { import = "lazyvim.plugins.extras.util.mini-hipatterns" },
    { import = "lazyvim.plugins.extras.ui.mini-animate" },
    { import = "lazyvim.plugins.extras.coding.copilot" },
    { import = "lazyvim.plugins.extras.util.project" },
    -- import/override with your plugins
    { import = "plugins" },
  },
  ui = {
    icons = {
      loaded = "",
      not_loaded = "",
    },
  },

  defaults = {
    -- By default, only LazyVim plugins will be lazy-loaded. Your custom plugins will load during startup.
    -- If you know what you're doing, you can set this to `true` to have all your custom plugins lazy-loaded by default.
    lazy = true,
    -- It's recommended to leave version=false for now, since a lot the plugin that support versioning,
    -- have outdated releases, which may break your Neovim install.
    version = false, -- always use the latest git commit
    -- version = '*', -- try installing the latest stable version for plugins that support semver
  },
  install = { colorscheme = { "tokyonight", "habamax" } },
  checker = { enabled = true }, -- automatically check for plugin updates
  performance = {
    rtp = {
      -- disable some rtp plugins, reference from NvChad
      disabled_plugins = {
        "2html_plugin",
        "ftplugin",
      },
    },
  },
})


```

**Reply by dpetka2001** [reply]
You should put those in a file in `~/.config/nvim/lua/plugins/` directory as stated in the [docs](https://www.lazyvim.org/configuration#-file-structure). For disabling all LazyVim default keymaps (those that exist in the Lazyvim `keymaps.lua` that you can find [here](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/keymaps.lua), not the ones that are configured inside a plugin spec via the `keys` section or the LSP ones), you could do 
```lua
-- ~/.config/nvim/lua/plugins/core.lua
return {
  {
    "LazyVim/LazyVim",
    opts = {
      defaults = {
	    keymaps = true, -- lazyvim.config.keymaps
      },
    },
  },
    {
    "neovim/nvim-lspconfig",
    init = function()
      local keys = require("lazyvim.plugins.lsp.keymaps").get()
      keys[#keys + 1] = { "ca", vim.lsp.buf.code_action, desc = "Code Action", mode = { "n", "v" }, } 
    end,
  },
}
```
